autoconfig.json

```diff
@@ -1,34 +1,119 @@
+          "/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ",
+          "/ip4/104.131.131.82/udp/4001/quic-v1/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ"
+        ]
+      },
+      "DelegatedConfig": {
+        "Read": [
+          "/routing/v1/providers",
+          "/routing/v1/peers",
+          "/routing/v1/ipns"
+        ],
+        "Write": []
+      }
+    },
+    "IPNI": {
+      "URL": "https://cid.contact",
+      "Description": "Network Indexer - content routing database for large storage providers",
+      "DelegatedConfig": {
+        "Read": [
+          "/routing/v1/providers"
+        ],
+        "Write": []
+      }
+    },
+    "IPNS": {
+      "URL": "https://specs.ipfs.tech/ipns/ipns-record/",
+      "Description": "InterPlanetary Name System - signed mutable records that point to CIDs with optional metadata",
+      "DelegatedConfig": {
+        "Read": [
+          "/routing/v1/ipns"
+        ],
+        "Write": [
+          "/routing/v1/ipns"
+        ]
+      }
+    },
+    "DNS": {
+      "URL": "https://datatracker.ietf.org/doc/html/rfc8484",
+      "Description": "DNS over HTTPS (DoH) - protocol for performing DNS resolution via the HTTPS protocol",
+      "DelegatedConfig": {
+        "Read": [
+          "/dns-query"
+        ],
+        "Write": []
+      }
+    },
+    "Example": {
+      "URL": "https://example.com",
+      "Description": "Example routing system demonstrating extensibility. Clients should ignore unknown APIs they don't recognize",
+      "DelegatedConfig": {
+        "Read": [
+          "/example/v0/read"
+        ],
+        "Write": [
+          "/example/v0/write"
+        ]
+      }
+    }
+  },
   "DNSResolvers": {
     "eth.": [
       "https://dns.eth.limo/dns-query",
       "https://dns.eth.link/dns-query"
     ]
   },
-  "DelegatedRouters": {
-    "mainnet-for-nodes-with-dht": [
-      "https://cid.contact/routing/v1/providers"
-    ],
-    "mainnet-for-nodes-without-dht": [
-      "https://delegated-ipfs.dev/routing/v1/providers",
-      "https://delegated-ipfs.dev/routing/v1/peers",
-      "https://delegated-ipfs.dev/routing/v1/ipns"
-    ]
+  "DelegatedEndpoints": {
+    "https://cid.contact": {
+      "Systems": ["IPNI"],
+      "Read": [
+        "/routing/v1/providers"
+      ],
+      "Write": []
+    },
+    "https://delegated-ipfs.dev": {
+      "Systems": ["AminoDHT", "IPNI", "IPNS", "DNS"],
+      "Read": [
+        "/routing/v1/providers",
+        "/routing/v1/peers",
+        "/routing/v1/ipns",
+        "/dns-query"
+      ],
+      "Write": [
+        "/routing/v1/ipns"
+      ]
+    },
+    "https://example.com": {
+      "Systems": ["Example"],
+      "Read": [
+        "/example/v0/read"
+      ],
+      "Write": [
+        "/example/v0/write"
+      ]
+    }
   },
-  "DelegatedPublishers": {
-    "mainnet-for-ipns-publishers-with-http": [
-      "https://delegated-ipfs.dev/routing/v1/ipns"
-    ]
+  "Profiles": {
```

Comment:
Who are the "profiles" here meant to be seen by / interacted with that it's abstracting complexity? For example:
1. End users where the implementers have given them flexibility to choose their profile
2. Implementers who want an easy way to figure out what to enable / disable for their implementation

e.g. does the browser profile really want to be here vs in some defaults file chosen by the implementers and that can be different whether the platform is nodejs vs browser?

FWIW I do like the concept of the profiles and suggestions for implementers my question is if this seems like it'll be valuable to implementers spec'd in the file vs as some prose next to it / on the website.

Reply:
In my mind it was always closer to (2) – implementers. In places like Kubo, end user would never use these, but have something that implementers created.

To make this more clear, I've removed `Profiles`  from JSON and instead put dome docs in `index.html` (9746ad1e5c4049b1abe8175f9a970a267ea7cbed).